lib/rubocop/cop/performance/zip_without_block.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Performance
+      # Checks for `map { |id| [id] }` and suggests replacing it with `zip`.
+      #
+      # @safety
+      #   This cop is unsafe for novel definitions of `map` and `collect`
+      #   on non-Enumerable objects that do not respond to `zip`.
+      #   To make your object enumerable, define an `each` method
+      #   as described in https://ruby-doc.org/core/Enumerable.html
+      #
+      # @example
+      #   # bad
+      #   [1, 2, 3].map { |id| [id] }
+      #
+      #   # bad
+      #   [1, 2, 3].map { [_1] }
+      #
+      #   # good
+      #   [1, 2, 3].zip
+      #
+      #   # good (no offense)
+      #   [1, 2, 3].map { |id| id }
+      #
+      #   # good (no offense)
+      #   [1, 2, 3].map { |id| [id, id] }
```

Comment:
I think the following example is neither lacking nor excessive for user :-)

```suggestion
      # @example
      #   # bad
      #   [1, 2, 3].map { |id| [id] }
      #
      #   # good
      #   [1, 2, 3].zip
```